Automattic/wp-calypso
# Activity Log: Convert ActivityCard component to functional+TypeScript

Part of a continuing effort to migrate Activity Log components from a class-based to a functional structure.

#### Changes proposed in this Pull Request

* Convert `ActivityCard` from a class-based component to a functional one.
* Convert `ActivityCard` from JavaScript to TypeScript.
* Add several more properties to the `Activity` type, in the context of `ActivityCard`.

#### Testing instructions

***Full regression testing:*** This pull request is not intended to add any new behavior or changes to the appearance of the page. For more information on expected behavior, see #54420, #54626, #54695, #54759, and #54904.

To properly test:

* Start Calypso Green and visit the Activity Log. (These refactors do not extend to the Calypso Blue experience.)
* Check that all types of Activity card behave exactly as they do in production. Be sure to investigate cards that can be expanded, cards with media, and cards belonging to both simple sites and multisites.
